ls1intum/Artemis
# `Quiz exercises`: Do not include ungraded results on scores page

### Checklist
#### General
#### Server

### Motivation and Context
Graded submission of quiz exercises were shadowed by not-graded submissions made during practise period after the end of the exercise period. 
Fixes #7150.

### Description
The reason was that the server was always returning the latest submission. This PR changes database query so that it looks only for graded results.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Quiz Exercise

1. As student, solve the quiz and wait for your grade (screenshot 1)
1. As instructor, verify the student has a grade and then releases the quiz for practice
1. As student, resolves again the quiz
1. As instructor, verify that the original grade is still present.

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ParticipationResource.java | 94% | ✅              |

### Screenshots

![](https://user-images.githubusercontent.com/88991954/266276023-a01aa87a-7b93-4677-ba8e-6e6c2a36eb4c.png)
